web-registry/src/hooks/projects/useProjectWithMetadata.ts

```diff
@@ -0,0 +1,167 @@
+  isEdit,
+  onChainProject,
+  navigateNext,
+  projectEditSubmit,
+  anchored = true,
+}: Props): Res => {
+  let metadata: any;
+  let offChainProject: OffChainProject | undefined;
+  const reactQueryClient = useQueryClient();
+  const [updateProject] = useUpdateProjectByIdMutation();
+
+  // In project creation mode, we query the off-chain project since there's no on-chain project yet.
+  // In this case, the router param projectId is the off-chain project uuid.
+  const create = !!projectId && !isEdit;
+  const { data: projectByIdRes } = useQuery(
+    getProjectByIdQuery({
+      client: graphqlClient,
+      enabled: create,
+      id: projectId,
+    }),
+  );
+  const projectById = projectByIdRes?.data?.projectById;
+  if (create && projectById) {
+    offChainProject = projectById;
+    metadata = projectById.metadata;
+  }
+
+  // In project edit mode, we can query the on-chain (anchored) project metadata.
+  // In this case, the router param projectId is the on-chain project id.
+  const edit = !!projectId && isEdit;
+  // Metadata
+  const { data: anchoredMetadata } = useQuery(
+    getMetadataQuery({
+      iri: onChainProject?.metadata,
+      enabled: !!onChainProject?.metadata && edit && anchored,
+    }),
+  );
+  const { data: projectByOnChainIdRes } = useQuery(
+    getProjectByOnChainIdQuery({
+      client: graphqlClient,
+      enabled: edit,
+      onChainId: projectId as string,
+    }),
+  );
+  if (edit) {
+    offChainProject = projectByOnChainIdRes?.data?.projectByOnChainId;
+    if (anchored) {
+      metadata = anchoredMetadata;
+    } else {
+      metadata = offChainProject?.metadata;
+    }
+  }
+  const metadataReload = useCallback(async (): Promise<void> => {
+    if (create) {
+      await reactQueryClient.invalidateQueries({
+        queryKey: getProjectByIdKey(projectId),
+      });
+    }
+    if (edit) {
+      await reactQueryClient.invalidateQueries({
+        queryKey: getProjectKey(projectId),
+      });
+      await reactQueryClient.invalidateQueries({
+        queryKey: getProjectByOnChainIdKey(projectId),
+      });
+    }
+  }, [create, edit, reactQueryClient, projectId]);
+
+  const metadataSubmit = useCallback(
+    async (values: Values): Promise<void> => {
+      const newMetadata = { ...metadata, ...values };
+      try {
+        if (isEdit && anchored) {
+          await projectEditSubmit(newMetadata);
+        } else {
+          await updateProject({
+            variables: {
+              input: {
+                id: offChainProject?.id,
+                projectPatch: {
+                  metadata: newMetadata,
+                },
+              },
+            },
+          });
+          !isEdit && navigateNext();
+        }
+        await metadataReload();
+      } catch (e) {
+        // TODO: Should we display the error banner here?
+        // https://github.com/regen-network/regen-registry/issues/554
+        // eslint-disable-next-line no-console
+        console.log(e);
```

Comment:
```suggestion
setErrorCodeAtom(ERRORS.DEFAULT);
setErrorModalAtom(atom => (atom.description = String(error)));
```

Reply:
we don't always want to show the error modal here since we don't necessarily submit an on chain tx, and it will already get shown if needed through `projectEditSubmit`
I think in this case, we only want to `setErrorCodeAtom` if `!isEdit || !anchored` (ie if we don't submit a tx but just do a db update) but let's tackle this in a separate task https://github.com/regen-network/regen-registry/issues/554, this PR is already getting too large 😅